### ISSUE TITLE
DatePicker: correctly show hover range when end date is selected first

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-design-system#8454]
**monday.com sync:** #18048581149

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

When selecting a range end value at present time, you are not shown a hover range to select the start date.

### Acceptance Criteria

A hover range should be shown if a user is selecting the end date first just as the hover range is shown when selecting the start date.

### Relevant Info

https://github.com/Esri/calcite-design-system/pull/8444#pullrequestreview-1789327570

### Which Component

date-picker

### Example Use Case

https://codepen.io/AaronPutterman/pen/PoVrdoY?editors=100

1. select the end date input first
2. Select a date
3. Hover over other dates and notice nothing is displayed.
4. Reload the app
5. Select the start date input first
6. Select a date
7. Hover over other dates and see the range style appearing allowing you to select a range from a single date picker.

### Priority impact

p4 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by geospatialem** [comment]
Blocked by https://github.com/Esri/calcite-design-system/pull/8402.

**Comment by paulcpederson** [comment]
Looks like the blocking issue is resolved. We have an open issue in Online which I think is the same as this. Essentially you can't select a range correctly if you select the end date first. Additionally the current UI allows selection of a start date later than the end date in this scenario. 

@geospatialem happy to open a new issue if this isn't what you're trying to capture here.

**Comment by geospatialem** [comment]
> sue in Online which I think is the same as this. Essentially you can't select a range correctly if you select the end date first. Additionally the current UI allows selection of a start date later than the end date in this scenario.

Good catch, will get this one into triage. Believe your use case fits the above request, so no need to open a new issue. 

@paulcpederson Can you speak to the impact for Online to help the team prioritize the effort?

**Comment by paulcpederson** [comment]
Moderate impact I think. While this is a very common component in many of our pages (content, gallery, search, etc) it requires the user to specifically select the end date first to exercise it, so I don't think a lot of users are going to hit it.
